addon/components/inputs/datetime.js

```diff
@@ -57,8 +69,43 @@ export default AbstractInput.extend({
     return moment(value).format(DATE_TIME_FORMAT)
   },
 
+  // == Lifecycle Hooks =======================================================
+
+  init () {
+    this._super(...arguments)
+
+    const value = this.get('value')
+    const timezone = this.get('timezone')
+    let currentDateTime = getMomentTimezone(value, timezone)
+
+    if (!this.get('defaultToCurrentDateTime')) {
+      currentDateTime = value
+    } else {
+      currentDateTime = getFormattedTime(currentDateTime, this.get('dateTimeFormat'), timezone)
+    }
+
+    this.setProperties({
+      storedDateTimeValue: currentDateTime,
+      localTimezone: moment().format('Z')
+    })
+  },
+
   // == Actions ===============================================================
 
   actions: {
+    /**
+     * Sets the user's selected date and time to the bunsen model and stores it in the
+     * case that the user selects away from the date-time picker radio button but then returns
+     * @param {Object} value - the users selected date/time moment() object
+     * @returns {undefined}
+     */
+    selectDate (value) {
+      const timezone = this.get('timezone')
+      const datetime = getMomentTimezone(value, timezone, true)
+      const formattedTime = getFormattedTime(datetime, this.get('dateTimeFormat'), timezone)
+
+      this.set('storedDateTimeValue', formattedTime)
```

Comment:
This internal state will have the tendency to get out of date with the actual form's value. In future versions of Bunsen, we won't be re-rendering components needlessly so you can't rely on `init` to keep it in sync.